smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs

```diff
@@ -223,28 +257,70 @@ public virtual HttpResponse Request(HttpRequest smartsheetRequest)
                         restRequest.AddHeader(header.Key, header.Value);
                     }
                 }
-
+                Boolean bodyAdded = false;
                 if (smartsheetRequest.Entity != null && smartsheetRequest.Entity.GetContent() != null)
                 {
-                    restRequest.AddParameter(smartsheetRequest.Entity.ContentType, Util.ReadAllBytes(smartsheetRequest.Entity.GetBinaryContent()), 
-                        smartsheetRequest.Entity.ContentType, ParameterType.RequestBody);
+                    //restRequest.AddParameter(smartsheetRequest.Entity.ContentType, Util.ReadAllBytes(smartsheetRequest.Entity.GetBinaryContent()), 
+                    //    smartsheetRequest.Entity.ContentType, ParameterType.RequestBody);
+                    bodyAdded = true;
+                    if (smartsheetRequest.Entity.ContentType == "application/json") {
+                        restRequest = restRequest.AddStringBody(smartsheetRequest.Entity.GetContentAsString(), ContentType.Json);
+                        // JsonParameter param = new JsonParameter();
+                        // restRequest = restRequest.AddBody(param);
+                    } else {
+  
+                        BodyParameter param = new BodyParameter(smartsheetRequest.Entity.ContentType, Util.ReadAllBytes(smartsheetRequest.Entity.GetBinaryContent()), smartsheetRequest.Entity.ContentType);
+                        restRequest = restRequest.AddBody(param);
+                    }
+
                 }
 
                 // Set the client base Url.
-                httpClient.BaseUrl = new Uri(smartsheetRequest.Uri.GetLeftPart(UriPartial.Authority));
-
+                //httpClient.BaseUrl = new Uri(smartsheetRequest.Uri.GetLeftPart(UriPartial.Authority));
+                Uri baseUrlAsURI = new Uri(smartsheetRequest.Uri.GetLeftPart(UriPartial.Authority));
+                RestClientOptions options = new RestClientOptions(baseUrlAsURI);
+                options.FollowRedirects = true;
+                this.httpClient = new RestClient(options, null, null, true);
                 Stopwatch timer = new Stopwatch();
 
                 // Make the HTTP request
                 timer.Start();
-                restResponse = httpClient.Execute(restRequest);
+                Task<RestResponse> restResponseAsTask = this.httpClient.ExecuteAsync(restRequest);
+                restResponseAsTask.Wait();
+                restResponse = restResponseAsTask.Result;
                 timer.Stop();
 
                 LogRequest(restRequest, restResponse, timer.ElapsedMilliseconds);
 
                 if (restResponse.ResponseStatus == ResponseStatus.Error)
                 {
-                    throw new HttpClientException("There was an issue connecting.");
+                    Boolean restClientDebugging = false;
+                    if (restClientDebugging) {
```

Comment:
nitpick: this will work for now but a possible alternative would be conditional compilation directives if they make sense in this context: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives#conditional-compilation